### PR TITLE
#27 Add Track object type

### LINKED ISSUE #27: Implement Track data type
### Summary

We need a `Track` data type in the CP stack to maintain the confirmed tracks that a host vehicle needs for planning purposes. These `Track` types should maintain the state of the track (e.g. confirmed, tentative) as well as associate the `DetectedObject`s.

### Reasoning for new functionality

See above.

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This PR adds a tracked object type (`Track`) to represent detected objects that have persisted across several perception iterations.

## Related Issue

Closes #27

## Motivation and Context

Detected object represent entities perceived during the current perception iteration, but we need to represent and manage entities that have persisted across several iterations.

## How Has This Been Tested?

Nothing to test right now, this PR adds only a data type.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: include/cooperative_perception/track.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Leidos
```
Comment (MishkaMN) [inline include/cooperative_perception/track.hpp:2]: should say 2023

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: include/cooperative_perception/track.hpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * @tparam StateType State vector type for object's motion model
+ * @tparam CovarianceType Covariance matrix type for object's motion model
+ */
+template <typename StateType, typename CovarianceType>
+struct Track
+{
+  units::time::second_t timestamp;
+  StateType state;
+  CovarianceType covariance;
+  TrackStatus status;
```
Comment (MishkaMN) [inline include/cooperative_perception/track.hpp:57]: would we need some kind of id?

Comment (MishkaMN) [reply include/cooperative_perception/track.hpp:57]: discussed that this is particularly needed inside the object. Different class will handle the tracking or hash id.

Comment (adamlm) [reply include/cooperative_perception/track.hpp:57]: I thought the consensus was that we would hold off on adding an ID field until necessary. Or do you think we should add it now?

**Pull request merged**